image_client/picturae_csv_create.py

```diff
@@ -257,35 +260,16 @@ def taxon_concat(self, row):
 
         return str(gen_spec), str(full_name), str(first_intra), str(tax_name), str(hybrid_base)
 
-
-    def taxon_unmatch_create(self, unmatched_taxa: pd.DataFrame):
-        """taxon_unmatch_create: creates sql query for creating new records in taxa unmatch,
-                                from rows that did not pass TNRS successfully,
-                                either through spelling, or taxonomic errors.
-            args:
-                unmatched_taxa: a pandas dataframe with unmatched taxa terms filtered by score
-        """
-        print("uploading unmatched taxa")
-        for index, row in unmatched_taxa.iterrows():
-            catalognumber = unmatched_taxa.columns.get_loc("CatalogNumber")
-
-            sql = create_tnrs_unmatch_tab(row=row, df=unmatched_taxa, tab_name='taxa_unmatch')
-
-            sql_result = get_one_match(connection=self.specify_db_connection, tab_name='taxa_unmatch',
-                                      id_col='CatalogNumber', key_col='CatalogNumber',
-                                      match=row[catalognumber], match_type='integer')
-            if sql_result is None:
-                insert_table_record(connection=self.specify_db_connection, logger_int=self.logger, sql=sql)
-            else:
-                pass
-
     def taxon_check_real(self):
         """taxon_check_real:
            Sends the concatenated taxon column, through TNRS, to match names,
            with and without spelling mistakes, only checks base name
            for hybrids as IPNI does not work well with hybrids
            """
 
```

Comment:
add rpy2 import to function level , as rpy2 initializes upon import it would also run when picturae_importer was imported to clienttools , and would slow down the whole script.